docs/docs/mail.md

```diff
@@ -1,345 +1,180 @@
-timing them out, IP address changes, passing through one-way firewalls, require high-availability servers with 
-load balancing, require databases for stored state and so on.
-
-With a stream or RPC oriented approach the enclave would be expected to respond to bytes from a client more or less
-immediately, meaning it has to be always running and would have to persist uploaded data itself. Unfortunately
-persistence in enclaves is tricky due to the existence of [Memento attacks](security.md#memento-attacks).
-
-RPCs (including REST) can be hard to integrate with fully reliable systems that can tolerate restarts at any moment.
-Because server restarts break connections, in a loosely coupled multi-party system connection-oriented protocols
-require expensive HA load balancing to ensure users aren't often exposed to errors caused by planned restarts. Even
-then it isn't possible to eliminate such errors entirely, just reduce their impact. Load balancing in an enclave
-environment requires special support because the client is intimately aware of the exact *CPU* the enclave is
-running on - thus switching a client between servers isn't transparent. 
-
-!!! note
-    Future versions of Conclave are planned to add support for enclave clustering. 
-
-All this suggests the primary paradigm exposed to enclaves should be messages. The host will take care of
-delivering messages and providing atomic, transactional semantics over them so many kinds of enclaves won't need an
-encrypted database at all. The untrusted host can also take on the significant burden of moving data around,
-storing it to disk, sorting it, applying backpressure, exposing to the admin if there are backlogs etc.
+### Side-channel attacks
+
+A malicious host can infer crucial information if it knows the size or timing of a message.
+Conclave guards against these types of [side-channel attacks](security.md#side-channel-attacks) as follows.
+
+Conclave pads Mail items to a uniform size. This blocks any attempt to infer the contents of messages based on the 
+precise length of a message. By default, Conclave Mail uses a moving average size to pad messages. However, you can 
+configure the size of your application's messages to a reasonable upper limit.
+
+To avoid the host guessing information from message timing, you can send empty Mail items even when you have nothing 
+to say. For example, if an enclave is running an auction between users, and you wish to hide who won, the enclave 
+can send a "you won" or "you lost" message to every client.
+
+## Comparison to a classical public key infrastructure
+
+Conclave uses Conclave Mail to connect clients to enclaves because of the following reasons:
+
+1. Conclave Mail can work with a small [Trusted Computing Base (TCB)](https://en.wikipedia.org/wiki/Trusted_computing_base) 
+   when compared to HTTPS. A small TCB removes several common security problems at the origin.
```

Comment:
Compared to HTTPS, Conclave mail has a relatively small [Trusted Computing Base (TCB)](...). The small TCB helps to harden enclaves against potential [zero-day](https://en.wikipedia.org/wiki/Zero-day_(computing)) exploits that may exist in dependency libraries.

Reply:
Done.